### PR TITLE
Feature: Update summary page design

### PR DESCRIPTION
**Pull request by hibadjebabria169**
##  Context
see https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/75

#### Before
image

#### After
image

### Add metrics evolution popup
image

### Add all metadata table popup
image
### Add Ontology relation chart
image

### CODE REVIEW

**Comment by Bilelkihal** [comment]
@syphax-bouazzouni 

This is the design for the tabs:

**HTML:**
```
  <div class="ontology-details-tabs">
      <div class="tab-items">
          <div class="active">
              Summary
              <hr>
          </div>
          <div>
              Concepts
              <hr>
          </div>
          <div>
              Properties
              <hr>
          </div>
          <div>
              Notes
              <hr>
          </div>
          <div>
              Mappings
              <hr>
          </div>
          <div>
              Schemes
              <hr>
          </div>
          <div>
              Collections
              <hr>
          </div>
          <div>
              Widgets
              <hr>
          </div>
      </div>
      <hr>
  </div>
```
**CSS**

```
.ontology-details-tabs{
    margin: 40px 0;
}
.ontology-details-tabs > hr{
    border: 1px solid #f3f3f3;
    width: 100%;
}
.ontology-details-tabs .tab-items{
    display: flex;
    margin: 0 50px;
}
.ontology-details-tabs .tab-items div{
    font-size: 16px;
    font-weight: 400;
    color: #5e5e5e;
    margin-right: 50px;
    cursor: pointer;
    opacity: 60%;
    transition: opacity 0.3s ease;
}

.ontology-details-tabs .tab-items div:hover{
 
    opacity: 100%;
}

.ontology-details-tabs .tab-items div hr{
    display: none;
    margin-top: 10px;
}
.ontology-details-tabs .tab-items .active{
    font-weight: 500;
    color: black;
    opacity: 100%;
}
.ontology-details-tabs .tab-items .active hr{
    display: block;
    border: 1px solid #43D112;
    opacity: 100%;
}
```

File: yarn-error.log
Comment (syphax-bouazzouni) [inline yarn-error.log:1]: remove this file

File: app/views/ontologies/sections/_metadata.html.haml [hunk 1]
```diff
@@ -1,189 +1,132 @@
...
+  .summary-page-center.flex-column
+    .summary-page-container
+      .summary-page-sub-container
+        .summary-page-first-row
+          = render partial: 'ontologies/sections/metadata/ontology_description_section'
+          = render partial: 'ontologies/sections/metadata/ontology_fairness_section'
+          = render partial: 'ontologies/sections/metadata/ontology_relations_network'
+          = render partial: 'ontologies/sections/metadata/ontology_submissions_section'
+          = render partial: 'ontologies/sections/metadata/ontology_metrics_section'
+        .summary-page-second-row
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:11]: extract this section "depiction" to a helper or partial

File: app/views/ontologies/sections/_metadata.html.haml [hunk 2]
```diff
@@ -1,189 +1,132 @@
...
+          - depiction_url = @submission_latest.depiction
+          - unless depiction_url.nil?
+            .right_metadata
+              .depiction
+                %img{src: depiction_url}
+                .loop_icon
+                  - img = "<img src='#{asset_path("summary/loop.svg")}'/>"
+                  = link_to_modal(img.html_safe, "/ajax/submission/show_depiction?depiction_url=#{depiction_url}",data: { show_modal_title_value: "Depiction", show_modal_size_value: 'modal-xl' })          
+          .right_metadata
+            .depiction
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:21]: use here an existent component

File: app/views/ontologies/sections/_metadata.html.haml [hunk 3]
```diff
@@ -1,189 +1,132 @@
...
+              .caintainer_vertical
+                .projects_card
+                  %p Identifiers
+                  %img{src: asset_path("summary/info.svg")}/
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:25]: implement the info component

File: app/views/ontologies/sections/_metadata.html.haml [hunk 4]
```diff
@@ -1,189 +1,132 @@
...
+                %p.description_text_identifers
+                  VersionIRI
+              %p.normal_text
+                = render LinkFieldComponent.new(value: @submission_latest.versionIRI)
+              .creation_info
+                %p.description_text_identifers
+                  Identifier
+              %p.normal_text
+                = render LinkFieldComponent.new(value: @submission_latest.identifier)
+          - properties = {Creation_date: @submission_latest.creationDate, Modification_date: @submission_latest.modificationDate}
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:41]: put this in the controller

File: app/views/ontologies/sections/_metadata.html.haml [hunk 5]
```diff
@@ -1,189 +1,132 @@
...
+          = render DropdownContainerComponent.new(title: 'Dates', id: 'dates') do
+            - properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s) do
+                = render DateTimeFieldComponent.new(value: v) 
+          - properties = {creators: @submission_latest.hasCreator, contributors: @submission_latest.hasContributor, curators: @submission_latest.curatedBy,
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:46]: put this in the controller

File: app/views/ontologies/sections/_metadata.html.haml [hunk 6]
```diff
@@ -1,189 +1,132 @@
...
+                          = render ChipButtonComponent.new(text: v, type: "static",'data-controller':' tooltip', title: v, class:'text-truncate', style: 'max-width: 200px; display: block')
 
-    %div.right-hand-content
 
-      -# Misc links pane
-      %section.ont-metadata-card.ont-links-card
-        %div.ont-section-toolbar
-          %header.pb-2.font-weight-bold= t('ontology_details.metadata.links')
-        %a{:href => "#{(@submission_latest || @ontology).id}?display=all", :target => '_blank', :class => "btn btn-primary"}= t('ontology_details.metadata.go_to_rest_api_json_entry')
+          - properties = {Is_Format_Of: @submission_latest.isFormatOf, Has_Format: @submission_latest.hasFormat,
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:61]: put this in the controller

File: app/views/ontologies/sections/_metadata.html.haml [hunk 7]
```diff
@@ -1,189 +1,132 @@
...
-          %header.pb-2.font-weight-bold= "#{t('ontology_details.metadata.projects_using')} #{@ontology.acronym}"
-          = link_to(new_project_path(), "aria-label": t('ontology_details.metadata.create_new_project'), title: t('ontology_details.metadata.create_new_project')) do
-            %i.fas.fa-lg.fa-plus-circle{"aria-hidden": "true", style: "margin-left: 0.5rem"}
-        - if @projects.empty?
-          %p.font-italic= "#{t('ontology_details.metadata.no_projects_using')} #{@ontology.acronym}"
-        - else
-          %div.border-top
-            - for project in @projects
-              %p= link_to(project.name, project_path(project.acronym))
+          - properties = {known_Usage: @submission_latest.knownUsage, designed_For_Ontology_Task:@submission_latest.designedForOntologyTask,
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:67]: put this in the controller

File: app/views/ontologies/sections/_metadata.html.haml [hunk 8]
```diff
@@ -1,189 +1,132 @@
...
+                    %p.clickable_text= project.name
+            - properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s, value: Array(v).join(','))
+            = render FieldContainerComponent.new(label: 'includedInDataCatalog') do
+              - Array(@submission_latest.includedInDataCatalog).each do |catalog|
+                - data_catalog = $DATA_CATALOG_VALUES.select{ |cat_uri, cat_label|  catalog[cat_uri] }.first || [catalog, catalog]
+                %div.m-1.my-3
+                  = render ChipButtonComponent.new(text: data_catalog[1], url: catalog, type: "clickable")
 
+          - properties = {Accrual_method: @submission_latest.accrualMethod, Accrual_periodicity: @submission_latest.accrualPeriodicity, Accrual_policy: @submission_latest.accrualPolicy,
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:92]: put this in the controller

File: app/views/ontologies/sections/_metadata.html.haml [hunk 9]
```diff
@@ -1,189 +1,132 @@
...
+          Knowledge_representation_paradigm: @submission_latest.conformsToKnowledgeRepresentationParadigm, Engineering_methodology: @submission_latest.usedOntologyEngineeringMethodology,  Created_With: @submission_latest.usedOntologyEngineeringTool,
+          Competency_question: @submission_latest.competencyQuestion}
+          = render DropdownContainerComponent.new(title: 'Methodology', id: 'methodology') do
+            - properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s, value: Array(v).join(',')) 
+          .right_metadata
+            .depiction
+              .vists_image
+                %p Visits
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:101]: extract this in a helper

File: app/views/ontologies/sections/_metadata.html.haml [hunk 10]
```diff
@@ -1,189 +1,132 @@
...
+                - if visits_data(@ontology)
+                  = link_to(@ontology.links["analytics"] + "?apikey=#{get_apikey}&format=csv", "aria-label": "Download as CSV", title: "Download as CSV") do
+                    %img{alt: "", src: asset_path("summary/arrow_down.svg")}
+              = render partial: "visits"
+          = render SummarySectionComponent.new(title: 'Get my metadata back')  do
+            - unless @submission_latest.nil?
+              %div{data:{controller: 'metadata-downloader'}}
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:108]: extract this in a partial or helper

File: app/views/ontologies/sections/_metadata.html.haml [hunk 11]
```diff
@@ -1,189 +1,132 @@
...
+                =javascript_include_tag "jsonld"
+                %button{:id => "getMetadataBackNquadsBtn", :class => "btn btn-primary", 'data-action': 'metadata-downloader#downloadNQuads'} N-Triple
+                %button{:id => "getMetadataBackJsonldBtn", :class => "btn btn-primary", 'data-action': 'metadata-downloader#downloadJsonLd'} Json-LD
+                %button{:id => "getMetadataBackXmlBtn", :class => "btn btn-primary", 'data-action': 'metadata-downloader#downloadXML'} RDF/XML
+          .Summary-page-card
+            - unless @ontology.view?
+              %div.ont-section-toolbar
+                %header.pb-2.font-weight-bold= "Views of #{@ontology.acronym}"
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:116]: extract this section into partial

File: app/views/ontologies/sections/metadata/_ontology_description_section.html.haml [hunk 12]
```diff
@@ -0,0 +1,63 @@
+.Summary-page-card
+  %h4.summary-page-card-title General information
+  .account-page-info-column.align-items-center.description_text
+    = render TextAreaFieldComponent.new(value: @submission_latest.description)
+    - unless @submission_latest.logo.nil?
+      %img{height: "85", src: @submission_latest.logo, width: "189"}/
+  .creation_info
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/metadata/_ontology_description_section.html.haml:7]: use component here

File: app/views/ontologies/sections/_licenses.html.haml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+= turbo_frame_tag 'application_modal_content' do
+  - @licenses.each do |attribute|
+    - attr = attribute_infos(attribute)
+    - value = Array(@submission_latest.send(attribute))
+    .creation_info
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_licenses.html.haml:5]: use component here

File: app/views/layouts/ontology_viewer/_header.html.haml [hunk 14]
```diff
@@ -2,33 +2,37 @@
...
+    .ontology-details-header-left-container{style:'width: 80%'}
+      .ontology-details-name-bar
+        %div{style:'max-width: 60%'}
+          = @ontology.name
+          %span
+            = "("+ @ontology.acronym+")"
+        - unless sub.hasLicense.nil?
+          .ontology-details-licence
+            %div
+              - value = attribute_enforced_values('hasLicense')[sub.hasLicense] || sub.hasLicense
```
Comment (syphax-bouazzouni) [inline app/views/layouts/ontology_viewer/_header.html.haml:23]: use componenent

File: app/views/layouts/ontology_viewer/_header.html.haml [hunk 15]
```diff
@@ -2,33 +2,37 @@
...
-      = link_to(edit_ontology_path(@ontology.acronym), "aria-label": "Edit ontology details", title: "Edit ontology details") do
-        %i.fas.fa-lg.fa-user-edit
\ No newline at end of file
+        .ontology-details-last-update
+          %img{src: asset_path("update.svg")}/
+          %div
+            = t('ontology_details.header.last_uploaded')
+            = l(Date.parse(sub.creationDate), format: :monthfull_day_year)
+    .ontology-details-header-right-container.justify-content-end{style: 'min-width: 20%;'}
+      %a.mx-1.ontology-details-edit-button{href: "#{(@submission_latest || @ontology).id}?display=all", :target => '_blank'}
```
Comment (syphax-bouazzouni) [inline app/views/layouts/ontology_viewer/_header.html.haml:33]: use component

File: app/views/layouts/ontology_viewer/_header.html.haml [hunk 16]
```diff
@@ -2,33 +2,37 @@
...
+        = inline_svg_tag "json.svg"
+      - if @ontology.admin?(session[:user])
+        %a.mx-1.ontology-details-edit-button{href: edit_ontology_path(@ontology.acronym)}
```
Comment (syphax-bouazzouni) [inline app/views/layouts/ontology_viewer/_header.html.haml:36]: use component

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
I marked all the changes that remain to do in the code.

@hibadjebabria169 please see and fix.

File: app/assets/images/summary/logo.png
[the remark this answers is not in the document]
Comment (hibadjebabria169) [reply app/assets/images/summary/logo.png:1]: I couldn't delete this one, i'll double check

File: yarn-error.log
Comment (hibadjebabria169) [reply yarn-error.log:1]: Capture d’écran du 2023-07-06 11-33-25

File: app/views/ontologies/sections/_metadata.html.haml [hunk 17]
```diff
@@ -1,189 +1,44 @@
...
+        .summary-page-first-row
+          = render partial: 'ontologies/sections/metadata/ontology_description_section'
+          = render partial: 'ontologies/sections/metadata/ontology_fairness_section'
+          = render partial: 'ontologies/sections/metadata/ontology_relations_network'
+          = render partial: 'ontologies/sections/metadata/ontology_submissions_section'
+          = render partial: 'ontologies/sections/metadata/ontology_metrics_section'
+        .summary-page-second-row
+          = render partial: 'ontologies/sections/metadata/ontology_depiction_section'
+          = render partial: 'ontologies/sections/metadata/ontology_identifiers_section'
+          = render DropdownContainerComponent.new(title: 'Dates', id: 'dates') do
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:14]: extract this to a partial

File: app/views/ontologies/sections/_metadata.html.haml [hunk 18]
```diff
@@ -1,189 +1,44 @@
...
+            - @persons_organizations_properties.each do |k, values|
+              %p.field-normal_text
+              = render FieldContainerComponent.new(label: k.to_s) do
+                %div.d-flex.flex-wrap
+                  - Array(values).each do |val_str|
+                    - val_str.split(',').each do |v|
+                      - if v
+                        %div.m-1
+                          = render ChipButtonComponent.new(text: v, type: "static",'data-controller':' tooltip', title: v, class:'text-truncate', style: 'max-width: 200px; display: block')
+          = render DropdownContainerComponent.new(title: 'Links', id: 'link') do
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:28]: extract this to a partial

File: app/views/ontologies/sections/_metadata.html.haml [hunk 19]
```diff
@@ -1,189 +1,44 @@
...
+            - @links_properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s, value: v)
+          = render partial: 'ontologies/sections/metadata/ontology_all_projects_section'
+              
+          = render DropdownContainerComponent.new(title: 'Methodology', id: 'methodology') do
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:33]: extract this to a partial

File: app/views/ontologies/sections/_metadata.html.haml [hunk 20]
```diff
@@ -1,189 +1,44 @@
...
+            - @methodology_properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s, value: Array(v).join(',')) 
+          = render partial: 'ontologies/sections/metadata/ontology_visits_section'
+          = render SummarySectionComponent.new(title: 'Get my metadata back')  do
+            - unless @submission_latest.nil?
+              %div{data:{controller: 'metadata-downloader'}}
+                =javascript_include_tag "jsonld"
+                %button{:id => "getMetadataBackNquadsBtn", :class => "btn btn-primary", 'data-action': 'metadata-downloader#downloadNQuads'} N-Triple
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:41]: make this button bigger and take all the spaces

File: app/views/ontologies/sections/metadata/_ontology_identifiers_section.html.haml [hunk 21]
```diff
@@ -0,0 +1,24 @@
+.right_metadata
+    .depiction
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/metadata/_ontology_identifiers_section.html.haml:2]: use components here

File: app/views/ontologies/sections/metadata/_ontology_projects_section.html.haml [hunk 22]
```diff
@@ -0,0 +1,32 @@
+%p.normal_text
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/metadata/_ontology_projects_section.html.haml:1]: use components here

Comment (syphax-bouazzouni) [reply app/views/ontologies/sections/metadata/_ontology_projects_section.html.haml:1]: and extract an array of (label, values) as you did for the properties.

File: app/views/ontologies/sections/_metadata.html.haml [hunk 23]
```diff
@@ -1,189 +1,44 @@
...
+          = render partial: 'ontologies/sections/metadata/ontology_submissions_section'
+          = render partial: 'ontologies/sections/metadata/ontology_metrics_section'
+        .summary-page-second-row
+          = render partial: 'ontologies/sections/metadata/ontology_depiction_section'
+          = render partial: 'ontologies/sections/metadata/ontology_identifiers_section'
+          = render DropdownContainerComponent.new(title: 'Dates', id: 'dates') do
+            - @dates_properties.each do |k, v|
+              = render FieldContainerComponent.new(label: k.to_s) do
+                = render DateTimeFieldComponent.new(value: v) 
+          = render DropdownContainerComponent.new(title: 'Persons and organizations', id: 'person_and_organization') do
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/_metadata.html.haml:18]: extract to a partial this

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**